File: 김금환/스택, 큐/기능개발.cpp
#include <vector>
#include <cmath>

using namespace std;

vector<int> workingDay;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    
    for (int i = 0; i < progresses.size(); ++i) {
        workingDay.push_back(ceil(((double)100 - progresses[i]) / speeds[i]));
    }
        
    int cnt = 1;
    int day = workingDay[0];
    for (int i = 0; i < workingDay.size() - 1; ++i) {
        if (day >= workingDay[i + 1]) {
            ++cnt;
        } else {
            answer.push_back(cnt);
            cnt = 1;
            day = workingDay[i + 1];
        }
    }
    
    answer.push_back(cnt);
    
    return answer;
}

File: 이예진/스택, 큐/기능개발.py
import math
def solution(progresses, speeds):
    answer = []
    l = []
    # 며칠 간 작업을 해야하는지
    for i in range(len(progresses)):
        l.append(math.ceil((100-progresses[i])/speeds[i]))
    # tmp는 배포 개수에 대한 누적 변수, idx는 최근 것 중 가장 큰 작업일이 필요한 작업의 인덱스
    tmp = 1
    idx = 0
    for i in range(1,len(l)):
        # 최근 것 중에 큰 작업일이 필요하다면 이것도 같이 배포 가능
        if l[idx] >= l[i]:
            tmp+=1
        # 아니라면 그에 맞추어 초기화
        else:
            idx = i
            answer.append(tmp)
            tmp = 1
    # 마지막 배포 반영
    answer.append(tmp)        
    return answer